client/signup/main.jsx

```diff
@@ -135,14 +140,18 @@ const Signup = React.createClass( {
 		this.recordStep();
 	},
 
-	componentWillReceiveProps( { stepName } ) {
+	componentWillReceiveProps( { signupDependencies, stepName } ) {
 		if ( this.props.stepName !== stepName ) {
 			this.recordStep( stepName );
 		}
 
 		if ( stepName === this.state.resumingStep ) {
 			this.setState( { resumingStep: undefined } );
 		}
+
+		if ( this.dependenciesContainCartItem( signupDependencies ) ) {
+			this.setState( { hasCartItems: true } );
```

Comment:
`componentWillReceiveProps` is not called for first render, so `hasCartItems` will never be true on initial render. Just checking that that is ok?